src/Modules/CurrentMapModule/Services/CurrentMapService.cs

```diff
@@ -0,0 +1,71 @@
+﻿using EvoSC.Common.Database.Repository.Maps;
+using EvoSC.Common.Interfaces;
+using EvoSC.Common.Services.Attributes;
+using EvoSC.Common.Services.Models;
+using EvoSC.Manialinks.Interfaces;
+using EvoSC.Modules.Official.CurrentMapModule.Interfaces;
+using GbxRemoteNet.Events;
+using ISO3166;
+using Microsoft.Extensions.Logging;
+
+namespace EvoSC.Modules.Official.CurrentMapModule.Services;
+
+[Service(LifeStyle = ServiceLifeStyle.Singleton)]
+public class CurrentMapService : ICurrentMapService
+{
+    private readonly ILogger<CurrentMapService> _logger;
+    private readonly IManialinkManager _manialinkManager;
+    private readonly MapRepository _mapRepository;
+    private readonly IServerClient _client;
+
+    public CurrentMapService(IManialinkManager manialinkManager,
+        ILogger<CurrentMapService> logger,
+        MapRepository mapRepository, IServerClient client)
+    {
+        _logger = logger;
+        _manialinkManager = manialinkManager;
+        _mapRepository = mapRepository;
+        _client = client;
+    }
+
+    private static string GetCountry(string Zone)
+    {
+        var zones = Zone.Split("|");
+        if (zones.Length >= 2)
+        {
+            return zones[2];
+        }
+
+        return "Other";
+    }
+
+    public async Task ShowWidgetAsync()
+    {
+        _logger.LogWarning("***Asd");
+        var map = await _client.Remote.GetCurrentMapInfoAsync();
+        var dbMap = await _mapRepository.GetMapByUidAsync(map.UId);
+        List<Country> countries = Country.List.ToList();
+        var country = countries.Find(country => country.Name == GetCountry(dbMap?.Author?.Zone ?? ""));
+
+        await _manialinkManager.SendPersistentManialinkAsync("CurrentMapModule.CurrentMapWidget",
+            new { map = dbMap, country = country?.ThreeLetterCode ?? "WOR" });
+        _logger.LogDebug("Showing widget");
```

Comment:
We should extract the part of showing the manialink in a separate method to avoid code duplication.

Reply:
done